### ISSUE TITLE
Update from 1.43.0 to 1.44.0 breaks Lint/UnusedMethodArgument with numbered block parameters

### ISSUE THREAD

**Issue by lorenzk** [rubocop/rubocop#11501]
After updating from 1.43.0 to 1.44.0, RuboCop fails to see that a method argument is indeed used.

--------

## Expected behavior

RuboCop should continue to ignore `Lint/UnusedMethodArgument` in both functions below.

## Actual behavior

RuboCop does not see how the `year` argument is being used with the `_1` numbered block parameter in the first function. Applying the autocorrect breaks this code.

The second function with a named block parameter `p` works as expected.

## Steps to reproduce the problem

```ruby
# test.rb

def test_numbered(year) # -> [Correctable] Lint/UnusedMethodArgument: Unused method argument - `year`.
  people.filter { _1.year == year }
end

def test_named(year) # -> works as expected
  people.filter { |p| p.year == year }
end
```

```bash
> bundle exec rubocop test.rb
Inspecting 1 file
W

Offenses:

test.rb:1:19: W: [Correctable] Lint/UnusedMethodArgument: Unused method argument - year. If it's necessary, use _ or _year as an argument name to indicate that it won't be used. If it's unnecessary, remove it. You can also write as test_numbered(*) if you want the method to accept any arguments but don't care about them.
def test_numbered(year) # -> [Correctable] Lint/UnusedMethodArgument: Unused method argument - `year`.
                  ^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

## RuboCop version

```
> bundle exec rubocop -V
1.44.0 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.1.2) [arm64-darwin21]
  - rubocop-performance 1.15.2
  - rubocop-rails 2.17.4
```

**Comment by koic** [comment]
This issue is a dup with https://github.com/rubocop/rubocop/issues/11494 and has been resolved. Please wait for the next release.

**Comment by koic** [comment]
Never mind! Thank you for your feedback!
